packages/ccd-js-gen/README.md

```diff
@@ -0,0 +1,484 @@
+```
+
+#### function `createUnchecked`
+
+Construct a module client for interacting with a smart contract module on chain.
+This function _skips_ the check of the smart contract module being deployed on chain and leaves it up to the caller to ensure this.
+
+**Parameter:** The function takes a gRPC client from '@concordium/web-sdk'.
+
+```typescript
+const grpcClient = ...; // Concordium gRPC client from '@concordium/web-sdk'.
+const myModule: MyModule.Type = MyModule.createUnchecked(grpcClient);
+```
+
+> To run the checks manually use `await MyModule.checkOnChain(myModule);`.
+
+#### const `moduleReference`
+
+Variable with the reference of the smart contract module.
+
+```typescript
+const ref = MyModule.moduleReference;
+```
+
+#### function `getModuleSource`
+
+Get the module source of the deployed smart contract module from chain.
+
+```typescript
+const myModule: MyModule.Type = ...; // Generated module client
+const moduleSource = await MyModule.getModuleSource(myModule);
+```
+
+#### type `<ContractName>Parameter`
+
+Type representing the parameter for when instantiating a smart contract. The type is name `<ContractName>Parameter` where `<ContractName>` is the smart contract name in Pascal case.
+
+_This is only generated when the schema contains init-function parameter type._
+
+#### function `instantiate<ContractName>`
+
+For each smart contract in module a function for instantiating new instance is produced.
+These are named `instantiate<ContractName>` where `<ContractName>` is the smart contract name in Pascal case.
+
+An example for a smart contract module containing a smart contract named `my-contract`, the function becomes `instantiateMyContract`.
+
+**Parameters**
+
+The function parameters are:
+
+- `moduleClient` The client of the on-chain smart contract module.
+- `transactionMetadata` Metadata related to constructing the transaction, such as energy and CCD amount to include.
+- `parameter` Parameter to provide the smart contract module for the instantiation.
+
+  _With schema type:_ If the schema contains type information for the parameter, a type for the parameter is generated and used for this function. The type is name `<ContractName>Parameter` where `<ContractName>` is the smart contract name in Pascal case.
+
+  _Without schema type:_ If no schema information is present, the function uses the generic `Parameter` from `@concordium/web-sdk`.
+
+- `signer` The keys of to use for signing the transaction.
+
+**Returns**: Promise with the hash of the transaction.
+
+```typescript
+const myModule: MyModule.Type = ...; // Generated module client.
+const transactionMeta = { ... }; // Transaction meta information.
+const parameter = ...; // Parameter to pass the smart contract init-function.
+const signer = ...; // The keys to use for signing the transaction.
+const transactionHash = await MyModule.instantiateMyContract(myModule, transactionMeta, parameter, signer);
```

Comment:
Same as above, it'd be useful to provide valid dummy data here.

Reply:
I now provide some dummy values for the transaction metadata, but decided not to do so for `signer` and `parameter`.
For `signer` I find it adding a lot of noise, and for `parameter` it will completely depend on the contract details, which I have added in the comment for parameter.